core/store/src/trie/update.rs

```diff
@@ -33,6 +33,31 @@ impl TrieUpdate {
             self.trie.get(&self.root, key).map(|x| x.map(DBValue::from_vec))
         }
     }
+
+    /// Get values in trie update for a set of keys.
+    // This function will commit changes. Need to be used with caution
+    pub fn get_prefix_changes<'a, T: Iterator<Item = &'a Vec<u8>>>(
+        &mut self,
+        prefixes: T,
+    ) -> Result<HashMap<Vec<u8>, HashMap<Vec<u8>, Option<Vec<u8>>>>, StorageError> {
+        if !self.prospective.is_empty() {
+            self.commit();
+        }
+        let mut res = HashMap::new();
+        for prefix in prefixes {
+            let mut prefix_key_value_change = HashMap::new();
+            for key in self.iter(prefix)? {
+                if let Some(value) = self.committed.get(&key) {
```

Comment:
go over all committed changes and check if they start with prefix.
also we can use BTreeMap and use `self.committed.range(prefix)`